docs/how_to/apphooks.rst

```diff
@@ -8,288 +10,186 @@ this case, you can create a normal django CMS page without any content of its
+attribute to the class that reflects the way you'd include the applications' URLs into your project.
 
+For example, if your application requires that your project's URLs do::
 
-Replace ``myapp._urls`` with the path to your applications ``urls.py``. Now edit
-a page and open the advanced settings tab. Select your new apphook under
-"Application". Save the page.
+    url(r'^myapp/', include('myapp.urls', app_name='myapp')),
 
-.. warning::
+then your ``MyApphook`` class should include::
 
-    Whenever you add or remove an apphook, change the slug of a page containing an apphook or the
-    slug if a page which has a descendant with an apphook, the server must restart to re-load the
-    URL caches.
+    app_name = "myapp"
 
-    If you have the :ref:`ApphookReloadMiddleware` (recommended) installed, the server will
-    restart automatically. Otherwise, you will need to restart it manually.
+If you fail to this, then any templates in the application that invoke URLs using the form ``{% url 'myapp:index' %}``
+or views that call (for example) ``reverse('myapp:index')`` will throw a ``NoReverseMatch`` error.
 
-    If you have the
+*Unless* the class that defines the apphook specifies an ``app_name``, it can be attached only to one page at a time.
+Attempting to apply it a second times will cause an error. See :ref:`multi_apphook` for more on having multiple apphook
+instances.
 
-    An apphook won't appear until it is published. Take note that this also
-    means all parent pages must also be published.
 
-.. note::
+Loading new and re-configured apphooks
+======================================
 
-    If at some point you want to remove this apphook after deleting the
-    ``cms_apps.py`` there is a cms management command called ``uninstall apphooks`` that
-    removes the specified apphook(s) from all pages by name. eg. ``manage.py cms
-    uninstall apphooks MyApphook``. To find all names for uninstallable apphooks
-    there is a command for this as well ``manage.py cms list apphooks``.
+Certain apphook-related changes require server restarts in order to be loaded.
 
-If you attached the app to a page with the url ``/hello/world/`` and the app has
-a ``urls.py`` that looks like this::
+Whenever you:
 
-    from django.conf.urls import url
-    from sampleapp.views import main_view, sample_view
+* add or remove an apphook
+* change the slug of a page containing an apphook or the slug of a page which has a descendant with an apphook
 
-    urlpatterns = [
-        url(r'^$', main_view, name='app_main'),
-        url(r'^sublevel/$', sample_view, name='app_sublevel'),
-    ]
+the server must restart to re-load the URL caches.
 
-The ``main_view`` should now be available at ``/hello/world/`` and the
-``sample_view`` has the URL ``/hello/world/sublevel/``.
+If you have the :ref:`ApphookReloadMiddleware` installed, which is recommended, the server will restart automatically.
+Otherwise, you will need to restart it manually.
```

Comment:
Its a little scary-sounding that it "the server will restart automatically" and its not really true. The MW merely re-initializes the url patterns.